core/src/main/java/io/grpc/internal/Channelz.java

```diff
@@ -226,6 +229,13 @@ public Builder setLastCallStartedMillis(long lastCallStartedMillis) {
         return this;
       }
 
+      /** Sets the listen sockets. */
+      public Builder setListenSockets(List<Instrumented<SocketStats>> listenSockets) {
+        Preconditions.checkNotNull(listenSockets);
+        this.listenSockets = Collections.unmodifiableList(listenSockets);
```

Comment:
nit: if the caller retained access to `listenSockets`, they could still mutate it.   Consider making a defensive copy.

Reply:
Made defensive copy